crates/weaver_semconv/src/v2/mod.rs

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! The new way we want to define data going forward.
+
+use std::collections::BTreeMap;
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    deprecated::Deprecated,
+    group::GroupSpec,
+    semconv::{Imports, SemConvSpecV1},
+    stability::Stability,
+    v2::{attribute::AttributeDef, entity::Entity, event::Event, metric::Metric, span::Span},
+    YamlValue,
+};
+
+pub mod attribute;
+pub mod entity;
+pub mod event;
+pub mod metric;
+pub mod span;
+
+/// Common fields we want on all major components of semantic conventions.
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct CommonFields {
+    /// A brief description of the span.
+    pub brief: String,
+    /// A more elaborate description of the span.
+    /// It defaults to an empty string.
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    pub note: String,
+    /// Specifies the stability of the span.
+    pub stability: Stability,
+    /// Specifies if the semantic convention is deprecated. The string
+    /// provided as description MUST specify why it's deprecated and/or what
+    /// to use instead. See also stability.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub deprecated: Option<Deprecated>,
```

Comment:
default and Option?

Reply:
Can you clarify what you mean by this comment?

Reply:
`#[serde(default)]` is not needed for Option fields. They deserialize to None by default anyway.